--- provider/postgresql.py	2021-12-01 00:00:00.000000000 +0000
+++ provider/postgresql.py	2021-12-01 00:00:00.000000000 +0000
@@ -133,7 +133,7 @@
             if self.properties:
                 result = [res for res in result if res[0] in self.properties]
             self.columns = SQL(', ').join(
-                [Identifier(item[0]) for item in result]
+                [Identifier(self.table, item[0]) for item in result]
                 )

             for k, v in dict(result).items():
@@ -207,7 +207,7 @@
         where_conditions = []
         if properties:
             property_clauses = [SQL('{} = {}').format(
-                Identifier(k), Literal(v)) for k, v in properties]
+                Identifier(self.table, k), Literal(v)) for k, v in properties]
             where_conditions += property_clauses
         if bbox:
             bbox_clause = SQL('{} && ST_MakeEnvelope({})').format(
@@ -302,7 +302,7 @@
             orderby = self._make_orderby(sortby) if sortby else SQL('')

             sql_query = SQL("DECLARE \"geo_cursor\" CURSOR FOR \
-             SELECT DISTINCT {} {} FROM {} {} {}").\
+             SELECT {} {} FROM {} {} {}").\
                 format(props,
                        geom,
                        Identifier(self.table),
